spark/src/main/scala/io/delta/tables/DeltaTable.scala

```diff
@@ -41,35 +41,30 @@ import org.apache.spark.sql.types.StructType
  * @since 0.3.0
  */
 class DeltaTable private[tables](
-    @transient private val _df: Dataset[Row],
-    @transient private val table: DeltaTableV2)
+    @transient private val _table: DeltaTableV2,
+    private val alias: Option[String] = None)
   extends DeltaTableOperations with Serializable {
 
-  protected def deltaLog: DeltaLog = {
-    /** Assert the codes run in the driver. */
-    if (table == null) {
-      throw DeltaErrors.deltaTableFoundInExecutor()
-    }
+  // Ensure table's initial snapshot has been captured, for consistent behavior.
+  _table.initialSnapshot
 
-    table.deltaLog
+  /** Assert the codes run in the driver. */
+  private def table: DeltaTableV2 = Option(_table).getOrElse {
+    throw DeltaErrors.deltaTableFoundInExecutor()
   }
 
-  protected def df: Dataset[Row] = {
-    /** Assert the codes run in the driver. */
-    if (_df == null) {
-      throw DeltaErrors.deltaTableFoundInExecutor()
-    }
+  protected def deltaLog: DeltaLog = table.deltaLog
 
-    _df
-  }
+  @transient
+  protected lazy val df: Dataset[Row] = alias.foldLeft(table.toDf)(_ as _)
 
   /**
    * Apply an alias to the DeltaTable. This is similar to `Dataset.as(alias)` or
    * SQL `tableName AS alias`.
    *
    * @since 0.3.0
    */
-  def as(alias: String): DeltaTable = new DeltaTable(df.as(alias), table)
+  def as(alias: String): DeltaTable = new DeltaTable(table, Some(alias))
```

Comment:
i am worried about this. Before one could do `dt.as("x").as("y")` and the resultant DF produced by it would have the both the aliases `x` and `y` in the plan, such that both `x.col1` and `y.col1` would have worked. Now it wont as the aliases are not stacked.. isnt it?

Reply:
As far as I knew, each subquery alias node blocks everything behind it. Once y is in scope, x shouldn't be accessible any more. If that is true, then it doesn't matter if those nodes stack or not. 

A quick scala spark test confirms:
```scala
spark.range(100).as("x").as("y").select("x.id")
```
fails with:
```
SparkException: [UNRESOLVED_COLUMN.WITH_SUGGESTION] A column or function parameter with name `x`.`id` cannot be resolved. Did you mean one of the following? [`y`.`id`].
```